logical/framework/path.go

```diff
@@ -89,7 +108,82 @@ type Path struct {
 	HelpDescription string
 }
 
-func (p *Path) helpCallback() OperationFunc {
+// OperationHandler defines and describes a specific operation handler.
+type OperationHandler interface {
+	Handler() OperationFunc
+	Properties() OperationProperties
+}
+
+// OperationProperties describes an operation for documentation, help text,
+// and other clients. A Summary should always be provided, whereas other
+// fields can be populated as needed.
+type OperationProperties struct {
+	// Summary is a brief (usually one line) description of the operation.
+	Summary string
+
+	// Description is extended documentation of the operation and may contain
+	// Markdown-formatted text markup.
+	Description string
+
+	// Examples provides samples of the expected request data. The most
+	// relevant example should be first in the list, as it will be shown in
+	// documentation that supports only a single example.
+	Examples []RequestExample
+
+	// Responses provides a list of response description for a given response
+	// code. The most relevant response should be first in the list, as it will
+	// be shown in documentation that only allows a single example.
+	Responses map[string][]Response
+
+	// Unpublished indicates that this operation should not appear in public
+	// documentation or help text. The operation may still have documentation
+	// attached that can be used internally.
+	Unpublished bool
+
+	// Deprecated indicates that this operation should be avoided.
+	Deprecated bool
+}
+
+// RequestExample is example of request data.
+type RequestExample struct {
+	Description string                 // optional description of the request
+	Value       map[string]interface{} // map version of sample JSON
```

Comment:
s/Value/Values? Or maybe s/Value/Params`?

Reply:
Or given the openapi field is `parameters` maybe `s/Value/Parameters`

Reply:
"Parameters" is very path-centric in OpenAPI, but since this is really and internal facing name that devs will see and edit, we could use that. I'm almost inclined to call it `Data`  since that's well aligned with `FieldData` which is what this is a sample of.

Reply:
Sure, that could work.